beacon-chain/forkchoice/doubly-linked-tree/store.go

```diff
@@ -251,3 +260,27 @@ func (s *Store) tips() ([][32]byte, []types.Slot) {
 	}
 	return roots, slots
 }
+
+// HighestReceivedBlockSlot returns the highest slot received by the forkchoice
+func (f *ForkChoice) HighestReceivedBlockSlot() types.Slot {
+	f.store.nodesLock.RLock()
+	defer f.store.nodesLock.RUnlock()
+	return f.store.highestReceivedSlot
```

Comment:
atomic unit64 instead?

Reply:
feels like an overkill, we do need to hold the nodes lock anyway